[OSDOCS#17058]: DC-2: Image Mirroring and oc mirror Tooling-CQA-2 (#115637)
openshift/openshift-docs: modules/olm-mirroring-catalog-airgapped.adoc

[id="olm-mirror-catalog-airgapped_{context}"]
= Mirroring catalog contents to airgapped registries

[role="_abstract"]

> If your mirror registry is on a completely disconnected, or airgapped, host, take the following actions.

Reviewer: Hmm maybe something like this to avoid self referential:

```suggestion
You can mirror catalog contents into airgapped registries.

If your mirror registry is on a completely disconnected, or airgapped, host, take the following actions.
```

[source,terminal]
----
$ oc adm catalog mirror \
    <index_image> \
    file:///local/index \
    -a ${REG_CREDS} \
    --insecure \
    --index-filter-by-os='<platform>/<arch>'
----
where:

`<index image>`:: Specifies the index image for the catalog that you want to mirror.
`file:///local/index`:: Specifies the content to mirror to local files in your current directory.
`-a ${REG_CREDS}`:: Optional: If required, specify the location of your registry credentials file.

> `--insecure`:: Optional: If you do not want to configure trust for the target registry, add the `--insecure` flag.

Reviewer: ```suggestion
`-a ${REG_CREDS}`:: Specifies the location of your registry credentials file. This is optional.
`--insecure`:: Specifies the `--insecure` flag. This flag is optional. If you do not want to configure trust for the target registry, add the `--insecure` flag.
```

> `<platform>/<arch>`:: Optional: Specify which platform and architecture of the index image to select when multiple variants are available. Images are specified as `'<platform>/<arch>[/<variant>]'`. This does not apply to images referenced by the index. Valid values are `linux/amd64`, `linux/ppc64le`, `linux/s390x`, `linux/arm64`, and `.*`

Reviewer: ```suggestion
`<platform>/<arch>`:: Specifies which platform and architecture of the index image to select when multiple variants are available. This is optional. Images are specified as `'<platform>/<arch>[/<variant>]'`. This does not apply to images referenced by the index. Valid values are `linux/amd64`, `linux/ppc64le`, `linux/s390x`, `linux/arm64`, and `.*`
```

--
+
--
.Example output

> +

Reviewer: This actually shows up as a + in the rendered preview, this should be removed:

```suggestion
```

[source,terminal]
----
$ oc adm catalog mirror \
    file://local/index/<repository>/<index_image>:<tag> \
    <mirror_registry>:<port>[/<repository>] \
    -a ${REG_CREDS} \
    --insecure \
    --index-filter-by-os='<platform>/<arch>'
----
+
where:

`file://`:: Specifies the `file://` path from the previous command output.
`<mirror_registry>:<port>[/<repository>]`:: Specifies the fully qualified domain name (FQDN) for the target registry to mirror the Operator contents to. The mirror registry `<repository>` can be any existing repository, or namespace, on the registry, for example `olm-mirror` as outlined in the prerequisites. If there is an existing repository found during mirroring, the repository name is added to the resulting image name. If you do not want the image name to include the repository name, omit the `<repository>` value from this line, for example `<mirror_registry>:<port>`.
`-a ${REG_CREDS}`:: Optional: If required, specifies the location of your registry credentials file.
`--insecure`:: Optional: If you do not want to configure trust for the target registry, add the `--insecure` flag.

> `--index-filter-by-os='<platform>/<arch>'`:: Optional: Specifies which platform and architecture of the index image to select when multiple variants are available. Images are specified as `'<platform>/<arch>[/<variant>]'`. This does not apply to images referenced by the index. Valid values are `linux/amd64`, `linux/ppc64le`, `linux/s390x`, `linux/arm64`, and `.*`

Reviewer: Reword this similarly to my suggestions earlier in this file

+
[NOTE]
====

> {quay} does not support nested repositories. As a result, running the `oc adm catalog mirror` command will fail with a `401` unauthorized error. As a workaround, you can use the `--max-components=2` option when running the `oc adm catalog mirror` command to disable the creation of nested repositories. For more information on this workaround, see "Unauthorized error thrown while using catalog mirror command with Quay registry".

Reviewer: Just a reminder to replace removed xrefs in procedure modules throughout your PR

$ oc adm catalog mirror \
    <mirror_registry>:<port>/<index_image> \
    <mirror_registry>:<port>[/<repository>] \
    --manifests-only \
    [-a ${REG_CREDS}] \
    [--insecure]
----
+
where:

> `--manifests-only`:: The `--manifests-only` flag is required for this step so that the command does not copy all of the mirrored content again.

Reviewer: ```suggestion
+
The `--manifests-only` flag is required for this step so that the command does not copy all of the mirrored content again.
```
